### PR TITLE
packages/storybook: use ts-loader for JS + resolve @backstage/core imports

### LINKED ISSUE #685: Storybook build failing in plugin PRs
## Expected Behavior

The storybook builds should not fail when there are no changes to storybook or it's dependencies.

## Current Behavior

Storybook verification build is failing with `Error: Can't resolve 'babel-loader' in '/home/runner/work/backstage/backstage/packages/storybook'`, see e.g. https://github.com/spotify/backstage/pull/651/checks?check_run_id=627548535

## Possible Solution

Might be an issue with the cache, might be an issue with deps sometimes not being built, or something else.

## Steps to Reproduce

? - Maybe this could be reproduced locally by adding a plugin, clean out the repo from dist and node_modules dirs, and then try to run tasks similar to what we're doing with the GitHub actions.

## Context

This also happened for the tech radar PR, where it was possibly fixed by first adding deps and then removing them. https://github.com/spotify/backstage/pull/628

## Your Environment

Build in GitHub Actions

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes #685, as well as broken master builds of storybook

### CODE REVIEW

**Comment by Rugvip** [comment]
This fix should also allow us to provide implementations of APIs in the storybook config, since the 2 versions of @backstage/core broke that before.

File: packages/storybook/.storybook/main.js [hunk 1]
```diff
@@ -12,36 +12,46 @@ module.exports = {
...
-        test: /\.(ts|tsx)$/,
-        use: [
-          {
-            loader: require.resolve('ts-loader'),
-            options: {
-              transpileOnly: true,
-            },
+    config.resolve.modules.push(coreSrc);
+
+    // Remove the default babel-loader for js files, we're using ts-loader instead
```
Comment (wesleyyee) [inline packages/storybook/.storybook/main.js:26]: so does this mean we also don't have polyfills for any of our stories?

**Review by wesleyyee — APPROVED** [review]
just a question

File: packages/storybook/.storybook/main.js (hunk 1 above)
Comment (Rugvip) [reply packages/storybook/.storybook/main.js:26]: We'll have the same as we had for TS I guess, so shouldn't change much

Comment (wesleyyee) [reply packages/storybook/.storybook/main.js:26]: yea es6 only

Comment (Rugvip) [reply packages/storybook/.storybook/main.js:26]: @wesleyyee [#655](https://github.com/spotify/backstage/pull/655/files#diff-a8d0e8b0feb26fb10e28b5d4739e7060) :grin:

**Pull request merged**